packages/firebase_remote_config/example/android/app/src/main/java/io/flutter/plugins/firebase/firebaseremoteconfigexample/MainActivity.java

```diff
@@ -1,13 +1,26 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
 package io.flutter.plugins.firebase.firebaseremoteconfigexample;
 
-import android.os.Bundle;
-import io.flutter.app.FlutterActivity;
-import io.flutter.plugins.GeneratedPluginRegistrant;
+import io.flutter.embedding.android.FlutterActivity;
+import io.flutter.embedding.engine.FlutterEngine;
+import io.flutter.embedding.engine.plugins.shim.ShimPluginRegistry;
+import io.flutter.plugins.firebase.core.FirebaseCorePlugin;
+import io.flutter.plugins.firebase.firebaseremoteconfig.FirebaseRemoteConfigPlugin;
 
 public class MainActivity extends FlutterActivity {
+
+  // TODO(cyanglaz): Remove this once v2 of GeneratedPluginRegistrant rolls to stable.
+  // https://github.com/flutter/flutter/issues/42694
   @Override
-  protected void onCreate(Bundle savedInstanceState) {
-    super.onCreate(savedInstanceState);
-    GeneratedPluginRegistrant.registerWith(this);
+  public void configureFlutterEngine(FlutterEngine flutterEngine) {
+    super.configureFlutterEngine(flutterEngine);
+    flutterEngine.getPlugins().add(new FirebaseRemoteConfigPlugin());
+
+    ShimPluginRegistry shimPluginRegistry = new ShimPluginRegistry(flutterEngine);
+    FirebaseCorePlugin.registerWith(
```

Comment:
ditto from other PR

Reply:
I might have missed some updates about the V2 embedding. I took a look of other firebase plugin migration PR and figured we don't have to use the `ShimPluginRegistry` to manually register the `firebase core` plugin anymore. And instead we need to add the e2e plugin. So I have updated the file the same. Could you please take another look see if it is good to merge? @xster

Reply:
I meant rather that https://github.com/FirebaseExtended/flutterfire/blob/master/packages/firebase_core/android/src/main/java/io/flutter/plugins/firebase/core/FirebaseCorePlugin.java has a v2 embedding implementation. Shouldn't it just work if you did `flutterEngine.getPlugins.add(new FirebaseCorePlugin());`?

Reply:
I see thanks! Will update the dependency as well.